### DISCUSSION TITLE
Can this library also be used to encode video frames and push frames to RTMP servers?

### DISCUSSION THREAD

**Discussion by sarthakz9** [ffmpeginteropx/FFmpegInteropX#299] [Q&A] [answered]
With the Mac FFmpeg terminal tool, I'm able to push a local video to RTMP servers using a command similar to this,

`ffmpeg -i file.mp4 -c:a aac -c:v libx264 -f flv -keyint_min 30 -g 30 -sc_threshold 0 -s 656x700 -r 15 -b:v 680k rtmp://domain/id`

Can I do the same using the FFmpegInteropX lib? or is it just a decoding library that can only be used to play content on devices not stream from them?

**Comment by djoij** [comment]
most probably not as it is mentioned that it is only a decoding library

**Comment by brabebhin** [comment, marked as answer]
If you want to stream data from a server and then encode it to a file, you can technically do this using some winRT APIs

The other way around, streaming to a server, I'm not so sure. I think you're the first to request such functionality.

**Reply by sarthakz9** [reply]
The use case is to stream live content from the app. FFmpeg itself supports pushing frames to an RTMP server, But I'm unsure if FFmpegInteropX lib exposes all the capabilities of FFmpeg lib or not. 
(Correct me If I'm wrong) FFmpegInteropX uses FFmpeg lib internally, does that imply FFmpeg lib itself is supported on UWP platforms as well?

**Reply by brabebhin** [reply]
Yes, we do use ffmpeg internally, but it is a custom build. Some features are disabled. We currently do not support the scenario you need. But you can fiddle around with the build scripts to enable some features.

**Reply by sarthakz9** [reply]
Thanks a lot for the help!
